runtime/src/tasks/fifo/tasks-fifo.c

```diff
@@ -881,8 +881,11 @@ void chpl_task_yield(void) {
 }
 
 
-void chpl_task_sleep(int secs) {
-  sleep(secs);
+void chpl_task_sleep(int time, double scale) {
+  struct timespec delay;
+  delay.tv_sec = (time_t)(scale*time);
+  delay.tv_nsec = (long)(1e3*(scale*time - floor(scale*time)));
```

Comment:
Oops, that's embarrassing. No wonder my Qthreads tests worked fine, but FIFO and MassiveThreads always seemed a bit off.